### PR TITLE
feat(node): Upgrade Node version to 16

### PR DESCRIPTION
**Pull request by trollepierre**
- [x] Change the comparator - previously the AppSections was NOT sorted alphabetically.

- [x] Figure should handle correctly separator: ","  because French LocaleString is used
Snapshots updated.

- [x] Fixed when yarn build:doc or yarn test : https://github.com/stylus/stylus/issues/2534#issuecomment-659414984

```
(node:74912) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:74912) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:74912) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
(node:74912) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(node:74912) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:74912) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
```

- [x] Fixed when yarn install, by running `yarn config set ignore-optional true`

```
warning Error running install script for optional dependency: "/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents: Command failed.
Exit code: 1
Command: node install
Arguments: 
Directory: /Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents
Output:
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.12.0
node-pre-gyp info using node@16.13.1 | darwin | x64
node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp info check checked for \"/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents/lib/binding/Release/node-v93-darwin-x64/fse.node\" (not found)
node-pre-gyp http GET https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.2.9/fse-v1.2.9-node-v93-darwin-x64.tar.gz
node-pre-gyp http 404 https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.2.9/fse-v1.2.9-node-v93-darwin-x64.tar.gz
node-pre-gyp WARN Tried to download(404): https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.2.9/fse-v1.2.9-node-v93-darwin-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for fsevents@1.2.9 and node@16.13.1 (node-v93 ABI, unknown) (falling back to source compile with node-gyp) 
node-pre-gyp http 404 status code downloading tarball https://fsevents-binaries.s3-us-west-2.amazonaws.com/v1.2.9/fse-v1.2.9-node-v93-darwin-x64.tar.gz 
gyp info it worked if it ends with ok
gyp info using node-gyp@5.1.1
gyp info using node@16.13.1 | darwin | x64
gyp info ok 
gyp info it worked if it ends with ok
gyp info using node-gyp@5.1.1
gyp info using node@16.13.1 | darwin | x64
gyp info find Python using Python version 2.7.18 found at \"/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python\"
(node:77462) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
(Use `node --trace-deprecation ...` to show where the warning was created)
gyp info spawn /System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python
gyp info spawn args [
gyp info spawn args   '/Users/trolle/projects/cozy/cozy-ui/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/trolle/projects/cozy/cozy-ui/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/trolle/Library/Caches/node-gyp/16.13.1/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/trolle/Library/Caches/node-gyp/16.13.1',
gyp info spawn args   '-Dnode_gyp_dir=/Users/trolle/projects/cozy/cozy-ui/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/trolle/Library/Caches/node-gyp/16.13.1/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.

No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.

gyp: No Xcode or CLT version detected!
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/Users/trolle/projects/cozy/cozy-ui/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Darwin 21.1.0
gyp ERR! command \"/Users/trolle/.nvm/versions/node/v16.13.1/bin/node\" \"/Users/trolle/projects/cozy/cozy-ui/node_modules/node-gyp/bin/node-gyp.js\" \"configure\" \"--fallback-to-build\" \"--module=/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents/lib/binding/Release/node-v93-darwin-x64/fse.node\" \"--module_name=fse\" \"--module_path=/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents/lib/binding/Release/node-v93-darwin-x64\" \"--napi_version=8\" \"--node_abi_napi=napi\" \"--napi_build_version=0\" \"--node_napi_label=node-v93\"
gyp ERR! cwd /Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents
gyp ERR! node -v v16.13.1
gyp ERR! node-gyp -v v5.1.1
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/Users/trolle/.nvm/versions/node/v16.13.1/bin/node /Users/trolle/projects/cozy/cozy-ui/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents/lib/binding/Release/node-v93-darwin-x64/fse.node --module_name=fse --module_path=/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents/lib/binding/Release/node-v93-darwin-x64 --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v93' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1064:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
node-pre-gyp ERR! System Darwin 21.1.0
node-pre-gyp ERR! command \"/Users/trolle/.nvm/versions/node/v16.13.1/bin/node\" \"/Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents/node_modules/node-pre-gyp/bin/node-pre-gyp\" \"install\" \"--fallback-to-build\"
node-pre-gyp ERR! cwd /Users/trolle/projects/cozy/cozy-ui/node_modules/fsevents
node-pre-gyp ERR! node -v v16.13.1
node-pre-gyp ERR! node-pre-gyp -v v0.12.0
node-pre-gyp ERR! not ok 
$ yarn release
```

### CODE REVIEW

File: react/AppSections/categories.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ export const sorter = (categoryA, categoryB) => {
     (categoryB.value === 'all' && 1) ||
     (categoryA.value === 'others' && 1) ||
     (categoryB.value === 'others' && -1) ||
-    categoryA.label > categoryB.label
+    categoryA.label.localeCompare(categoryB.label)
```
Comment (trollepierre) [inline react/AppSections/categories.js:51]: ??? this generates a change in Argos. I wonder if this is necessar
Capture d’écran 2022-01-03 à 14 07 31
y

Comment (trollepierre) [reply react/AppSections/categories.js:51]: https://github.com/cozy/cozy-ui/pull/1835/files#r777467457

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-10
+16
```
Comment (trollepierre) [inline .nvmrc:1]: is it a breaking change? 🤔

File: react/__snapshots__/examples.spec.jsx.snap [hunk 3]
```diff
@@ -52,58 +52,6 @@ exports[`ActionMenu should render examples: ActionMenu 1`] = `
 </div>"
 `;
 
-exports[`AppTitle should render examples: ActionMenu 1`] = `
```
Comment (JF-Cozy) [inline react/__snapshots__/examples.spec.jsx.snap]: ça a été introduit par le premier commit, on peut peut-être sqash d'une manière ou d'une autre ?

File: .nvmrc (hunk 2 above)
Comment (JF-Cozy) [reply .nvmrc:1]: je dirais que oui à cause du sort qui modifie l'ordre des choses

Comment (JF-Cozy) [reply .nvmrc:1]: je dirais que oui à cause du sort qui modifie l'ordre des choses

Comment (Crash--) [reply .nvmrc:1]: > is it a breaking change? 🤔

No, only code used in production is subject to breaking change. If you add "engine: 16" to the package.json, then it's a breaking change. Here it's not. 

> je dirais que oui à cause du sort qui modifie l'ordre des choses

It doesn't change the order on our production code. It'll have no impact on app calling UI. So no, it's not a BC.

File: react/__snapshots__/examples.spec.jsx.snap (hunk 3 above)
Comment (trollepierre) [reply react/__snapshots__/examples.spec.jsx.snap]: ok done

**Comment by trollepierre** [comment]
@Crash-- or @JF-Cozy : can I get one approval 🙏 ?

File: stylus/tools/mixins.styl [hunk 4]
```diff
@@ -79,7 +79,7 @@ hide()
 
     Weight: 3
 
-    Styleguide Tools.mixins.hide
+    Styleguide Tools.mixins.visually-hide
```
Comment (JF-Cozy) [inline stylus/tools/mixins.styl:82]: nit: devvrait être dans un autre commit je pense

**Review by JF-Cozy — APPROVED** [review]

**Pull request merged**